### ISSUE TITLE
resolveNestedSelectors:false doesn't prevent ampersand selectors from being evaluated for selector-class-pattern

### ISSUE THREAD

**Issue by HellGilo** [stylelint/stylelint#4948]
> Clearly describe the bug

Ampersand selectors are always parsed even when the default option for resolveNestedSelectors is false

> Which rule, if any, is the bug related to?

`selector-class-pattern`

> What code is needed to reproduce the bug?

e.g.

```scss

.Tag {
  &.no_answer {
    background-color: red;
  }
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
     "selector-class-pattern": ["^[A-Z][a-zA-Z0-9]+$", {
      "resolveNestedSelectors": false
    }],
  }
}
```

> Which version of stylelint are you using?

e.g. `13.7.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint filename.module.scss`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS nested properties.

> What did you expect to happen?

There shouldn't be any warning thrown for the nested class `.no_answer`

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
filename.module.scss
12:4    Expected class selector ".no_answer" to match specified pattern        selector-class-pattern
```

The fix is really simple: the if statement at selector-class-pattern/index.js line 60 should be reworked so that if `hasInterpolatingAmpersand(selector)` is true but `shouldResolveNestedSelectors` is false then the `parseSelector` step is skipped

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

I see that rule works correctly: `no_answer` doesn't match RegEx, because of underscore in a class name.

**Comment by HellGilo** [comment]
> Thanks for the report and for using the template.
> 
> I see that rule works correctly: `no_answer` doesn't match RegEx, because of underscore in a class name.

yeah, but no_answer is a nested selector. With resolveNestedSelectors set to false, the property shouldn't be evaluated against the rule

**Comment by hudochenkov** [comment]
I'm afraid there is misunderstanding from what this option does. It does not ignore nested selectors.

If we have SCSS:

```scss
.b {
    &__e { }
}
```

If `resolveNestedSelectors` is `false`, then rule will read second selector as `&__e`. If options set to `true` rule will understand it as `.b__e`. It will resolved it as preprocessor does.
